src/pages/Facility/settings/devices/components/DeviceServiceHistory.tsx

```diff
@@ -0,0 +1,132 @@
+import { useQuery } from "@tanstack/react-query";
+import { format } from "date-fns";
+import { Edit, Plus } from "lucide-react";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+
+import query from "@/Utils/request/query";
+import { ServiceHistoryResponse } from "@/types/device/device";
+import deviceApi from "@/types/device/deviceApi";
+
+import ServiceHistorySheet from "./ServiceHistorySheet";
+
+interface DeviceServiceHistoryProps {
+  facilityId: string;
+  deviceId: string;
+}
+
+export default function DeviceServiceHistory({
+  facilityId,
+  deviceId,
+}: DeviceServiceHistoryProps) {
+  const { t } = useTranslation();
+  const [isServiceSheetOpen, setIsServiceSheetOpen] = useState(false);
+  const [selectedServiceHistory, setSelectedServiceHistory] =
+    useState<ServiceHistoryResponse | null>(null);
+
+  const {
+    data: serviceHistory,
+    isLoading,
+    refetch,
+  } = useQuery({
+    queryKey: ["device", facilityId, deviceId, "serviceHistory"],
+    queryFn: query(deviceApi.serviceHistory.list, {
+      pathParams: {
+        facility_external_id: facilityId,
+        device_external_id: deviceId,
+      },
+    }),
+  });
+
+  const handleAddService = () => {
+    setSelectedServiceHistory(null);
+    setIsServiceSheetOpen(true);
+  };
+
+  const handleEditService = (service: ServiceHistoryResponse) => {
+    setSelectedServiceHistory(service);
+    setIsServiceSheetOpen(true);
+  };
+
+  const handleServiceSheetClose = () => {
+    setIsServiceSheetOpen(false);
+    refetch();
```

Comment:
use `invalidateQueries` instead

```suggestion
```

Reply:
Please do not resolve the conversation without addressing the feedback.